### PR TITLE
Add parameter to make transform publishing optional

### PR DESCRIPTION
**Pull request by afakihcpr**
It is useful in many use-cases to have the ability to run r_l without TF publising.

### CODE REVIEW

File: src/ros_filter.cpp [hunk 1]
```diff
@@ -1342,56 +1346,59 @@ namespace RobotLocalization
 
         // If the worldFrameId_ is the odomFrameId_ frame, then we can just send the transform. If the
         // worldFrameId_ is the mapFrameId_ frame, we'll have some work to do.
-        if (filteredPosition.header.frame_id == odomFrameId_)
```
Comment (afakihcpr) [inline src/ros_filter.cpp:1345]: All the following changes are indentation changes due to the addition of the if (publishTransform_)

**Comment by l0g1x** [comment]
+1

Krystian Gebis
University of Illinois - Chicago
Chicago EDT Robotics - Vice President
www.chicagoedt.org

**Comment by ayrton04** [comment]
Yeah, this is definitely a good idea. I've implemented this same thing more than once, but somehow it's never made its way back into the repo. I want to get Indigo and Jade released with the same source as Kinetic (just released yesterday), and then I'll merge this. Thanks for your help!

File: src/ros_filter.cpp (hunk 1 above)
Comment (ayrton04) [inline src/ros_filter.cpp:1345]: Thanks for the heads-up.

**Pull request merged**